    <!-- Research Section -->
    <section id="research" class="container content-section text-justify">
        <div class="row">
            <div class="col-lg-8 col-lg-offset-2">
                <h2>Research</h2>
                <p>The Fahlgren group at the Danforth Center uses and develops computational
                    approaches and infrastructure that leverage large datasets to address
                    biological problems. We emphasize the development of modular, reusable,
                    and open-source tools through collaborator- and community-driven efforts.
                    Our aim is to apply these tools to high-throughput genotyping and phenotyping
                    data to identify the genetic basis of traits in research model plants and
                    biofuel and food security crops.</p>
                <p>The ability to rapidly and non-destructively measure plant physical and
                    physiological features is a key bottleneck in plant research and breeding.
                    Imaging coupled with computer vision algorithms and statistical analysis are
                    a set of technologies that have the potential to address the plant phenotyping
                    bottleneck, but they introduce their own computing, interpretation, and data
                    management challenges that our group develops tools to address so that these
                    technologies can be utilized more broadly by the scientific community. Plant
                    Computer Vision (<a href="https://plantcv.danforthcenter.org">PlantCV</a>)
                    is our primary platform for developing a plant phenotyping toolbox. Through
                    PlantCV we are deploying computer vision, machine learning, and other data
                    science algorithms to extract biologically relevant data from image and sensor
                    datasets.</p>
                <p>A major emphasis of the Fahlgren group is collaboration, which enables us to
                    apply the tools we develop to a variety of plant systems. Diverse candidate
                    biofuel feedstocks such as Camelina sativa (oilseed) and Sorghum bicolor
                    (lignocellulosic feedstock) are major focuses in the group where we are
                    utilizing natural variation and high-throughput phenotyping to study the
                    genetic basis of traits that could improve these crops for bio-based fuels.
                    We are also developing tools for model systems (e.g. <i>Arabidopsis thaliana</i>
                    and <i>Setaria viridis</i>), food security crops (e.g. cassava), and other
                    systems for producing plant natural products (e.g. indigo).</p>
			</div>
		</div>
		<div class="row">
			<div class="col-lg-8 col-lg-offset-2">
				<h3>Ongoing Projects</h3>
				<div class="grant">
					<h4>PlantCV</h4>
					<b>Project homepage</b>: <a href="https://plantcv.danforthcenter.org">https://plantcv.danforthcenter.org</a><br>
					<b>Project summary:</b>: PlantCV is an image analysis software
					package for plant phenotyping. Our long-term goals for the PlantCV
					project are to: 1) Provide a common interface for a collection of
					image analysis techniques that are integrated from a variety of
					source packages and algorithms; 2) Utilize a modular architecture
					that enables flexibility in the design of analysis workflows and
					rapid assimilation and integration of new methods; 3) Develop openly
					in real-time in the cloud using an open-source framework to rapidly
					disseminate new methods and to encourage open contribution by
					stakeholders; and 4) Provide a simplified interface for users to
					utilize the underlying tools and build custom analysis workflows
					without significant experience with programming.
				</div>
				<div class="grant">
					<h4>Collaborative Research: CSSI: Framework: Data: Clowder Open
					Source Customizable Research Data Management, Plus-Plus</h4>
					<b>Funding agency</b>: National Science Foundation Office of Advanced
					Cyberinfrastructure (OAC)<br>
					<b>Program</b>: Cyberinfrastructure for Sustained Scientific
					Innovation (CSSI) - Data and Software<br>
					<b>Grant number</b>: <a href="https://www.nsf.gov/awardsearch/showAward?AWD_ID=1835543&HistoricalAwards=false">1835543</a><br>
					<b>Project homepage</b>: <a href="https://clowder.ncsa.illinois.edu">https://clowder.ncsa.illinois.edu</a><br>
					<b>Project summary</b>: Preserving, sharing, navigating, and reusing
					large and diverse collections of data is now essential to scientific
					discoveries in areas such as phenomics, materials science,
					geoscience, and urban science. These data navigation needs are also
					important when addressing the growing number of research areas where
					data and tools must span multiple domains. To support these needs
					effectively, new methods are required that simplify and reduce the
					amount of effort needed by researchers to find and utilize data,
					support community accepted data practices, and bring together the
					breadth of standards, tools, and resources utilized by a community.
					Clowder, an active curation based data management system, addresses
					these needs and challenges by distributing much of the data curation
					overhead throughout the lifecycle of the data, augmenting this with
					social curation and automated analysis tools, and providing
					extensible community-dependent means of viewing and navigating data.
					As an open source framework, built to be extensible at every level,
					Clowder is capable of interacting with and utilizing a variety of
					community tools while also supporting different data governance and
					ownership requirements.<br><br>
					The project enhances Clowder's core systems for the benefit of a
					larger group of users. It increases the level of interoperability
					with community resources, hardens the core software, and distributes
					core software development, while continuing to expand usage.
					Governance mechanisms and a business model are established to make
					Clowder sustainable, creating an appropriate governance structure to
					ensure that the software continues to be available, supportable, and
					usable. The effort engages a number of stakeholders, taking data from
					diverse but converging scientific domains already using the Clowder
					framework, to address broad interoperability and cross domain data
					sharing. The overall effort will transition the grassroots Clowder
					user community and Clowder's other stakeholders (such as current and
					potential developers) into a larger organized community, with a
					sustainable software resource supporting convergent research data
					needs.
				</div>
				<div class="grant">
					<h4>STTR Phase II: Novel Analysis Tools for Production of Higher
					Indican Yielding Plants for Bio-based Indigo</h4>
					<b>Funding agency</b>: National Science Foundation Industrial
					Innovation and Partnerships (IIP)<br>
					<b>Program</b>: Small Business Technology Transfer Program Phase II
					(STTR)<br>
					<b>Grant number</b>: <a href="https://www.nsf.gov/awardsearch/showAward?AWD_ID=1831949&HistoricalAwards=false">1831949</a><br>
					<b>Project summary</b>: This Small Business Technology Transfer
					(STTR) Phase II project will develop a new tool for the analysis of
					in situ indican precursor in indigo plants, which when combined with
					genomic analysis and genetic linkage mapping in selectively bred
					indigo crops, will lead to high indican yielding breeding parental
					lines and ultimately competitively price natural indigo dye.
					Additionally, characterization of genetic markers will accelerate
					further crop improvement by understanding and harnessing the genes
					crucial to indican synthesis and other aspects of significance to
					overall indigo yield. These advancements will benefit customers,
					denim mills, by leading to a more reliable, lower cost plant-derived
					indigo supply. The success of this multiphase STTR project will
					enable a cost-competitive, cleaner, and more sustainable denim dyeing
					process, while greatly expanding the market for domestically produced
					natural indigo. Commercialization of a more consistent and higher
					yielding US-grown indigo plant that produces high purity indigo
					powder can replace the current standard of synthetic, imported indigo
					powder. While in demand by the marketplace today, plant-derived
					indigo is currently only used in premium denim products due to the
					high cost per pound resulting from low yields per plant per acre. The
					direct result of this research will be to open new market segments
					and expand existing market penetration for US-grown and manufactured
					biobased indigo for the textile industry, an addressable market of
					$1.86B. The methods and technology developed through this project
					have a direct path to the commercial marketplace and the industry is
					ready to support biobased textile dyes such as plant-derived indigo.
					<br><br>
					During this project, reference genome resources will be constructed
					for indigo feedstock crops, F1 mapping populations will be
					constructed for <i>P. tinctoria</i>, <i>I. tinctoria</i>, and <i>I.
					suffruticosa</i> varieties, and design of the handheld rapid assay
					device will be validated through laboratory analysis. The reference
					genome for <i>I. suffruticosa</i> will be built using Pacific
					Biosciences SMRT sequencing and assembly. In addition to this
					reference genome, whole-genome resequencing will be performed on
					available <i>I. suffruticosa</i> and <i>I. tinctoria</i> varieties.
					Nucleotide variation and variant effect prediction will be made
					between the Indigofera-based indigo varieties. This variation will be
					used to develop markers to evaluate intervarietal crosses. Controlled
					greenhouse crosses of three species will be made to create F1 mapping
					populations for use in constructing a genetic map and linkage
					mapping, in combination with the resources developed. Intraspecies
					varieties exhibiting distinct phenotypic traits of commercial
					interest or notable dye yield differences will be selected for
					crossing to generate the F1 populations. Results from
					laboratory-based fluorometry equipment will be evaluated for efficacy
					and then compared against extractive indigo dye analysis from the
					leaf biomass. A final prototype will be constructed based on these
					findings and validated through use in the laboratory and in the field.
				</div>
				<div class="grant">
					<h4>Optimizing Tradeoffs Implicit During Bioenergy Crop Improvement:
					Understanding the Effect of Altered Cell Wall and Sugar Content on
					Sorghum-associated Pathogenic Bacteria</h4>
					<b>Funding agency</b>: US Department of Energy Office of Biological &
					Environmental Research (BER)<br>
					<b>Program</b>: Plant Feedstocks Genomics for Bioenergy<br>
					<b>Grant number</b>: <a href="https://pamspublic.science.energy.gov/WebPAMSExternal/Interface/Common/ViewPublicAbstract.aspx?rv=3dc7965f-9752-48ef-ab91-b280d82d1212&rtc=24&PRoleId=10">DE-SC0018072</a><br>
					<b>Project summary</b>: High-biomass-yielding crops may harbor
					modifications to cell walls, which are a major barrier to pathogen
					entry, and to the tissue distribution of sugars, which are the
					pathogen’s food source; hence they are likely to present previously
					unseen challenges for disease resistance. <i>Xanthomonas</i> is a
					known pathogen of sorghum (<i>Sorghum bicolor</i> (L.) Moench),
					though the incidence and impact of the disease has historically been
					low. We are working to establish the sorghum – <i>Xanthomonas</i>
					pathosystem as a model for deducing how latent microbial pathogens
					might exploit key biofuel crop traits. Our approach will be to
					quantitatively model the disease triangle that describes sorghum,
					pathogenic bacteria, and the environment. Field and laboratory
					experiments will be combined to determine bacterial susceptibility of
					genetically diverse sorghum genotypes that differ in cell wall and
					sugar composition. Standard plant pathology techniques combined with
					powerful phenomics approaches will provide a holistic view of this
					pathosystem within variable environments. Further, transcriptomics
					will be employed to elucidate mechanisms used by bacterial pathogens
					to induce sorghum susceptibility. Microbial pathogens are known to
					manipulate the sugar and cell wall characteristics of their hosts.
					Consequently, these characteristics will be analyzed during pathogen
					invasion. This research will reveal the mechanisms underlying
					tolerance to pathogens that must be maintained during biofuel trait
					optimization. The proposed research will yield a detailed
					understanding of the impact of bioenergy relevant traits on pathogen
					susceptibility. This is a necessary first step towards the
					development of novel routes for disease control that can be deployed
					in parallel with targeted alterations to sugar and cell wall
					composition during bioenergy crop improvement and breeding efforts.
				</div>
				<div class="grant">
					<h4>Genomics and Phenomics to Identify Yield and Drought Tolerance
					Alleles for Improvement of Camelina as a Biofuel Crop</h4>
					<b>Funding agency</b>: US Department of Agriculture National
					Institute of Food and Agriculture<br>
					<b>Program</b>: Plant Feedstocks Genomics for Bioenergy/Agriculture
					and Food Research Initiative (AFRI)<br>
					<b>Grant number</b>: <a href="https://portal.nifa.usda.gov/web/crisprojectpages/1010241-genomics-and-phenomics-to-identify-yield-and-drought-tolerance-alleles-for-improvement-of-camelina-as-a-biofuel-crop.html">2016-67009-25639</a><br>
					<b>Project summary</b>: Plant oils represent an outstanding potential
            		source of energy-dense hydrocarbons that can be used for fuels and
            		industrial raw materials, but a major challenge is to produce these
            		oils in non-food oilseed crops that have high yields and can grow
            		under marginal and varied climatic conditions. In recent years,
            		<i>Camelina sativa</i> has received considerable attention as a
            		potential non-food biofuels crop, but significant challenges remain
            		to develop stable, high-yielding, geographically adapted germplasm
            		suitable for biofuels production. We will utilize advanced
            		high-throughput phenotyping and genomics-based approaches to discover
            		useful gene/alleles controlling seed yield and oil content and
            		quality in camelina under water-limited conditions, and will identify
            		high-yielding cultivars suitable for production in different
            		geographical regions. The project includes three primary objectives:
            		1) Develop and apply automated, non-destructive high-throughput
            		phenotyping (HTP) protocols to evaluate the phenotypic diversity and
            		stress tolerance of a camelina panel consisting of 250 accessions,
            		grown under well-watered and water-limited conditions. 2) Discover
            		alleles/genes controlling morphological, physiological, seed, and oil
            		yield properties using genome-wide association studies (GWAS). 3)
            		Identify, test, and validate useful germplasm, including transgenic
            		lines producing drop-in ready jet fuel components, under diverse
            		environments and marginal production areas. Taken together, this
            		project will significantly advance the utilization of non-food
            		oilseed crops for biofuel production and provide guidance and insight
            		for future studies of phenomics-based crop improvement.
				</div>
				<div class="grant">
					<h4>The Missouri Transect: Climate, Plants, and Community</h4>
					<b>Funding agency</b>: National Science Foundation Office of
					Integrative Activities<br>
					<b>Program</b>: Established Program to Stimulate Competitive Research
					(EPSCoR) Research Infrastructure Improvement Program Track-1:
					(RII Track-1)<br>
					<b>Grant number</b>: <a href="https://www.nsf.gov/awardsearch/showAward?AWD_ID=1355406">IIA-1355406</a><br>
					<b>Project homepage</b>: <a href="https://missouriepscor.org">https://missouriepscor.org</a><br>
					<b>Project summary</b>: The Missouri Transect is a five-year effort to
					build infrastructure, knowledge, and collaborations in research and
					education across Missouri. The research and education activities are
					focused on understanding, modeling, and predicting 1) short- and
					long-term trends in temperature and water availability; 2) the impact
					of these trends on the productivity of native flora and agriculture
					crops; and 3) how different stakeholder communities are likely to
					respond to a changing climate. We have assembled interdisciplinary
					teams who will focus on specific areas of research and education and
					interact collaboratively to build the research platform across the
					state.<br><br>
					<b>Mission</b>: The Missouri Transect will enhance our state’s
					capacity to model and respond to the effects of climate change on
					plants and community on a local scale.<br><br>
					<b>Vision</b>: Missouri EPSCoR will enhance the state’s infrastructure
					for science and technology, stimulating Missouri’s economy and leading
					to job creation.<br><br>
					Members of the Climate, Plant, and Community Teams bring together
					their diverse research strengths and approaches to bear on three
					overlapping research questions: How do short- and long-term
					climate-scale trends affect crops and the natural flora? What genes
					are important to plant responses to these environmental variations?
					and How resilient are Missouri communities to variations in the
					climate and its impact on plants?<br><br>
					The teams assembled have a range of expertise, experience, and
					capabilities in research, education, diversity, outreach, and
					commercialization. Team members are drawn from participating
					institutions and are leaders in their respective fields.<br><br>
					Expected outcomes from this project include a strong climate
					monitoring infrastructure with the ability to predict daily, seasonal,
					annual and future variability; drought-tolerant crops; a regional
					resilience framework to help Missouri communities respond to climate
					change; modernization of Missouri’s cyberinfrastructure capable of
					handling “Big Data”; and a more technologically advanced workforce.
				</div>
			</div>
        </div>
        <div class="row">
        	<div class="col-lg-8 col-lg-offset-2 text-center">
        		<a href="#contact" class="btn btn-circle page-scroll">
            		<i class="fa fa-angle-double-down animated"></i>
        		</a>
        	</div>
        </div>
    </section>
