    <!-- Research Section -->
    <section id="publications" class="container content-section text-justify">
        <div class="row publications-section">
            <div class="col-lg-8 col-lg-offset-2">
                <h2>Publications</h2>
                <p>Zheng X, Carrington JC, Fahlgren N, Abbasi A, Berry JC. 2018.
                Antiviral functions of ARGONAUTE proteins during Turnip Crinkle Virus
                infection revealed by image-based trait analysis in Arabidopsis.
                <i>bioRxiv</i>:487322. DOI: <a href="https://doi.org/10.1101/487322">
                10.1101/487322</a>.</p>
                <p>Berry JC, Fahlgren N, Pokorny AA, Bart RS, Veley KM. 2018. An
                automated, high-throughput method for standardizing image color profiles
                to improve image-based plant phenotyping. <i>PeerJ</i> 6:e5727. DOI:
                <a href="https://doi.org/10.7717/peerj.5727">10.7717/peerj.5727</a>.</p>
				<p>Feldman MJ, Ellsworth PZ, Fahlgren N, Gehan MA, Cousins AB, Baxter I.
				2018. Components of water use efficiency have unique genetic signatures
				in the model C4 grass Setaria. <i>Plant Physiology</i> 178:699–715. DOI:
				<a href="https://doi.org/10.1104/pp.18.00146">10.1104/pp.18.00146</a>.</p>
                <p>Burnette M, Kooper R, Maloney JD, Rohde GS, Terstriep JA, Willis C,
                Fahlgren N, Mockler T, Newcomb M, Sagan V, Andrade-Sanchez P, Shakoor N,
                Sidike P, Ward R, LeBauer D. 2018. TERRA-REF Data Processing
                Infrastructure. In: <i>Proceedings of the Practice and Experience on
                Advanced Research Computing</i>. ACM, 27. DOI:
                <a href="https://doi.org/10.1145/3219104.3219152">
                10.1145/3219104.3219152</a>.</p>
				<p>Li H, Yin Z, Manley P, Burken JG, Shakoor N, Fahlgren N, Mockler T.
				2018. Early drought plant stress detection with bi-directional long-term
				memory networks. <i>Photogrammetric Engineering & Remote Sensing</i>
				84:459–468. DOI:
				<a href="https://doi.org/10.14358/PERS.84.7.459">
				10.14358/PERS.84.7.459</a>.</p>
				<p>Tovar JC, Hoyer JS, Lin A, Tielking A, Callen ST, Elizabeth Castillo
				S, Miller M, Tessman M, Fahlgren N, Carrington JC, Nusinow DA, Gehan MA.
				2018. Raspberry Pi-powered imaging for plant phenotyping. <i>Applications
				in Plant Sciences</i> 6:e1031. DOI:
				<a href="https://doi.org/10.1002/aps3.1031">10.1002/aps3.1031</a>.</p>
				<p>Gehan MA, Fahlgren N, Abbasi A, Berry JC, Callen ST, Chavez L, Doust
				AN, Feldman MJ, Gilbert KB, Hodge JG, Hoyer JS, Lin A, Liu S, Lizárraga
				C, Lorence A, Miller M, Platon E, Tessman M, Sax T. 2017. PlantCV v2:
				Image analysis software for high-throughput plant phenotyping.
				<i>PeerJ</i> 5:e4088. DOI: <a href="https://doi.org/10.7717/peerj.4088">
				10.7717/peerj.4088</a>.</p>
				<p>Fahlgren N, Bart R, Herrera-Estrella L, Rellán-Álvarez R, Chitwood DH,
				Dinneny JR. 2016. Plant scientists: GM technology is safe. <i>Science</i>
				351:824. DOI: <a href="https://doi.org/10.1126/science.351.6275.824-a">
				10.1126/science.351.6275.824-a</a>.</p>
				<p>Fahlgren N, Hill ST, Carrington JC, Carbonell A. 2016. P-SAMS: a web
				site for plant artificial microRNA and synthetic trans-acting small
				interfering RNA design. <i>Bioinformatics</i> 32:157–158. DOI:
				<a href="https://doi.org/10.1093/bioinformatics/btv534">
				10.1093/bioinformatics/btv534</a>.</p>
                <p>Abbasi A, Fahlgren N. 2016. Naïve Bayes pixel-level plant
                segmentation. In: <i>2016 IEEE Western New York Image and Signal
                Processing Workshop (WNYISPW)</i>. IEEE, 1–4. DOI:
                <a href="https://doi.org/10.1109/WNYIPW.2016.7904790">
                10.1109/WNYIPW.2016.7904790</a>.</p>
				<p>Wang H, Beyene G, Zhai J, Feng S, Fahlgren N, Taylor NJ, Bart R,
				Carrington JC, Jacobsen SE, Ausin I. 2015. CG gene body DNA methylation
				changes and evolution of duplicated genes in cassava. <i>Proceedings of
				the National Academy of Sciences of the United States of America</i>
				112:13729–13734. DOI: <a href="https://doi.org/10.1073/pnas.1519067112">
				10.1073/pnas.1519067112</a>.</p>
				<p>Fahlgren N, Feldman M, Gehan MA, Wilson MS, Shyu C, Bryant DW, Hill
				ST, McEntee CJ, Warnasooriya SN, Kumar I, Ficor T, Turnipseed S, Gilbert
				KB, Brutnell TP, Carrington JC, Mockler TC, Baxter I. 2015. A versatile
				phenotyping system and analytics platform reveals diverse temporal
				responses to water availability in Setaria. <i>Molecular Plant</i>
				8:1520–1535. DOI: <a href="https://doi.org/10.1016/j.molp.2015.06.005">
				10.1016/j.molp.2015.06.005</a>.</p>
				<p>Carbonell A, Fahlgren N, Mitchell S, Cox KL Jr, Reilly KC, Mockler TC,
				Carrington JC. 2015. Highly specific gene silencing in a monocot species
				by artificial microRNAs derived from chimeric miRNA precursors. <i>The
				Plant Journal: for cell and molecular biology</i> 82:1061–1075. DOI:
				<a href="https://doi.org/10.1111/tpj.12835">10.1111/tpj.12835</a>.</p>
				<p>Fahlgren N, Gehan MA, Baxter I. 2015. Lights, camera, action:
				high-throughput plant phenotyping is ready for a close-up. <i>Current
				Opinion in Plant Biology</i> 24:93–99. DOI:
				<a href="https://doi.org/10.1016/j.pbi.2015.02.006">
				10.1016/j.pbi.2015.02.006</a>.</p>
				<p>Garcia-Ruiz H, Carbonell A, Hoyer JS, Fahlgren N, Gilbert KB, Takeda
				A, Giampetruzzi A, Garcia Ruiz MT, McGinn MG, Lowery N, Martinez Baladejo
				MT, Carrington JC. 2015. Roles and programming of Arabidopsis ARGONAUTE
				proteins during Turnip Mosaic Virus infection. <i>PLoS Pathogens</i>
				11:e1004755. DOI: <a href="https://doi.org/10.1371/journal.ppat.1004755">
				10.1371/journal.ppat.1004755</a>.</p>
				<p>Gilbert KB, Fahlgren N, Kasschau KD, Chapman EJ, Carrington JC,
				Carbonell A. 2014. Preparation of multiplexed small RNA libraries from
				plants. <i>Bio-protocol</i> 4:e1275. DOI:
				<a href="https://doi.org/10.21769/BioProtoc.1275">
				10.21769/BioProtoc.1275</a>.</p>
				<p>Carbonell A, Takeda A, Fahlgren N, Johnson SC, Cuperus JT, Carrington
				JC. 2014. New generation of artificial microRNA and synthetic
				trans-acting small interfering RNA vectors for efficient gene silencing
				in Arabidopsis. <i>Plant Physiology</i> 165:15–29. DOI:
				<a href="https://doi.org/10.1104/pp.113.234989">
				10.1104/pp.113.234989</a>.</p>
				<p>Jeong D-H, Schmidt SA, Rymarquis LA, Park S, Ganssmann M, German MA,
				Accerbi M, Zhai J, Fahlgren N, Fox SE, Garvin DF, Mockler TC, Carrington
				JC, Meyers BC, Green PJ. 2013. Parallel analysis of RNA ends enhances
				global investigation of microRNAs and target RNAs of <i>Brachypodium
				distachyon</i>. <i>Genome Biology</i> 14:R145. DOI:
				<a href="https://doi.org/10.1186/gb-2013-14-12-r145">
				10.1186/gb-2013-14-12-r145</a>.</p>
				<p>Fahlgren N, Bollmann SR, Kasschau KD, Cuperus JT, Press CM, Sullivan
				CM, Chapman EJ, Hoyer JS, Gilbert KB, Grünwald NJ, Carrington JC. 2013.
				Phytophthora have distinct endogenous small RNA populations that include
				short interfering and microRNAs. <i>PloS ONE</i> 8:e77181. DOI:
				<a href="https://doi.org/10.1371/journal.pone.0077181">
				10.1371/journal.pone.0077181</a>.</p>
				<p>Carbonell A, Fahlgren N, Garcia-Ruiz H, Gilbert KB, Montgomery TA,
				Nguyen T, Cuperus JT, Carrington JC. 2012. Functional analysis of three
				Arabidopsis ARGONAUTES using slicer-defective mutants.
				<i>The Plant Cell</i> 24:3613–3629. DOI:
				<a href="https://doi.org/10.1105/tpc.112.099945">
				10.1105/tpc.112.099945</a>.</p>
				<p>Zhang C, Montgomery TA, Fischer SEJ, Garcia SMDA, Riedel CG, Fahlgren
				N, Sullivan CM, Carrington JC, Ruvkun G. 2012. The <i>Caenorhabditis
				elegans</i> RDE-10/RDE-11 complex regulates RNAi by promoting secondary
				siRNA amplification. <i>Current Biology</i>: CB 22:881–890. DOI:
				<a href="https://doi.org/10.1016/j.cub.2012.04.011">
				10.1016/j.cub.2012.04.011</a>.</p>
				<p>Fischer SEJ, Montgomery TA, Zhang C, Fahlgren N, Breen PC, Hwang A,
				Sullivan CM, Carrington JC, Ruvkun G. 2011. The ERI-6/7 helicase acts at
				the first stage of an siRNA amplification pathway that targets recent
				gene duplications. <i>PLoS Genetics</i> 7:e1002369. DOI:
				<a href="https://doi.org/10.1371/journal.pgen.1002369">
				10.1371/journal.pgen.1002369</a>.</p>
				<p>Hu TT, Pattyn P, Bakker EG, Cao J, Cheng J-F, Clark RM, Fahlgren N,
				Fawcett JA, Grimwood J, Gundlach H, Haberer G, Hollister JD, Ossowski S,
				Ottilar RP, Salamov AA, Schneeberger K, Spannagl M, Wang X, Yang L,
				Nasrallah ME, Bergelson J, Carrington JC, Gaut BS, Schmutz J, Mayer KFX,
				Van de Peer Y, Grigoriev IV, Nordborg M, Weigel D, Guo Y-L. 2011. The
				<i>Arabidopsis lyrata</i> genome sequence and the basis of rapid genome
				size change. <i>Nature Genetics</i> 43:476–481. DOI:
				<a href="https://doi.org/10.1038/ng.807">10.1038/ng.807</a>.</p>
				<p>Cuperus JT, Fahlgren N, Carrington JC. 2011. Evolution and functional
				diversification of <i>MIRNA</i> genes. <i>The Plant Cell</i> 23:431–442.
				DOI: <a href="https://doi.org/10.1105/tpc.110.082784">
				10.1105/tpc.110.082784</a>.</p>
				<p>Zhang C, Montgomery TA, Gabel HW, Fischer SEJ, Phillips CM, Fahlgren
				N, Sullivan CM, Carrington JC, Ruvkun G. 2011. mut-16 and other mutator
				class genes modulate 22G and 26G siRNA pathways in <i>Caenorhabditis
				elegans</i>. <i>Proceedings of the National Academy of Sciences of the
				United States of America</i> 108:1201–1208. DOI:
				<a href="https://doi.org/10.1073/pnas.1018695108">
				10.1073/pnas.1018695108</a>.</p>
				<p>Cuperus JT, Carbonell A, Fahlgren N, Garcia-Ruiz H, Burke RT, Takeda
				A, Sullivan CM, Gilbert SD, Montgomery TA, Carrington JC. 2010. Unique
				functionality of 22-nt miRNAs in triggering RDR6-dependent siRNA
				biogenesis from target transcripts in Arabidopsis. <i>Nature Structural &
				Molecular Biology</i> 17:997–1003. DOI:
				<a href="https://doi.org/10.1038/nsmb.1866">10.1038/nsmb.1866</a>.</p>
				<p>Fahlgren N, Jogdeo S, Kasschau KD, Sullivan CM, Chapman EJ, Laubinger
				S, Smith LM, Dasenko M, Givan SA, Weigel D, Carrington JC. 2010. MicroRNA
				gene evolution in <i>Arabidopsis lyrata</i> and
				<i>Arabidopsis thaliana</i>. <i>The Plant Cell</i> 22:1074–1089. DOI:
				<a href="https://doi.org/10.1105/tpc.110.073999">
				10.1105/tpc.110.073999</a>.</p>
				<p>International Brachypodium Initiative. 2010. Genome sequencing and
				analysis of the model grass <i>Brachypodium distachyon</i>. <i>Nature</i>
				463:763–768. DOI: <a href="https://doi.org/10.1038/nature08747">
				10.1038/nature08747</a>.</p>
				<p>Garcia-Ruiz H, Takeda A, Chapman EJ, Sullivan CM, Fahlgren N,
				Brempelis KJ, Carrington JC. 2010. Arabidopsis RNA-dependent RNA
				polymerases and Dicer-Like proteins in antiviral defense and small
				interfering RNA biogenesis during Turnip Mosaic Virus infection. <i>The
				Plant Cell</i> 22:481–496. DOI:
				<a href="https://doi.org/10.1105/tpc.109.073056">
				10.1105/tpc.109.073056</a>.</p>
				<p>Cuperus JT, Montgomery TA, Fahlgren N, Burke RT, Townsend T, Sullivan
				CM, Carrington JC. 2010. Identification of <i>MIR390a</i> precursor
				processing-defective mutants in Arabidopsis by direct genome sequencing.
				<i>Proceedings of the National Academy of Sciences of the United States
				of America</i> 107:466–471. DOI:
				<a href="https://doi.org/10.1073/pnas.0913203107">
				10.1073/pnas.0913203107</a>.</p>
				<p>Fahlgren N, Carrington JC. 2010. miRNA Target Prediction in Plants.
				In: Meyers BC, Green PJ eds. <i>Plant MicroRNAs</i>. Methods in Molecular
				Biology. Totowa, NJ: Humana Press, 51–57. DOI:
				<a href="https://doi.org/10.1007/978-1-60327-005-2_4">
				10.1007/978-1-60327-005-2_4</a>.</p>
				<p>Klevebring D, Street NR, Fahlgren N, Kasschau KD, Carrington JC,
				Lundeberg J, Jansson S. 2009. Genome-wide profiling of populus small
				RNAs. <i>BMC Genomics</i> 10:620. DOI:
				<a href="https://doi.org/10.1186/1471-2164-10-620">
				10.1186/1471-2164-10-620</a>.</p>
				<p>Gu W, Shirayama M, Conte D Jr, Vasale J, Batista PJ, Claycomb JM,
				Moresco JJ, Youngman EM, Keys J, Stoltz MJ, Chen C-CG, Chaves DA, Duan S,
				Kasschau KD, Fahlgren N, Yates JR 3rd, Mitani S, Carrington JC, Mello CC.
				2009. Distinct argonaute-mediated 22G-RNA pathways direct genome
				surveillance in the <i>C. elegans</i> germline. <i>Molecular Cell</i>
				36:231–244. DOI:
				<a href="https://doi.org/10.1016/j.molcel.2009.09.020">
				10.1016/j.molcel.2009.09.020</a>.</p>
				<p>Haas BJ, Kamoun S, Zody MC, Jiang RHY, Handsaker RE, Cano LM, Grabherr
				M, Kodira CD, Raffaele S, Torto-Alalibo T, Bozkurt TO, Ah-Fong AMV,
				Alvarado L, Anderson VL, Armstrong MR, Avrova A, Baxter L, Beynon J,
				Boevink PC, Bollmann SR, Bos JIB, Bulone V, Cai G, Cakir C, Carrington
				JC, Chawner M, Conti L, Costanzo S, Ewan R, Fahlgren N, Fischbach MA,
				Fugelstad J, Gilroy EM, Gnerre S, Green PJ, Grenville-Briggs LJ, Griffith
				J, Grünwald NJ, Horn K, Horner NR, Hu C-H, Huitema E, Jeong D-H, Jones
				AME, Jones JDG, Jones RW, Karlsson EK, Kunjeti SG, Lamour K, Liu Z, Ma L,
				Maclean D, Chibucos MC, McDonald H, McWalters J, Meijer HJG, Morgan W,
				Morris PF, Munro CA, O’Neill K, Ospina-Giraldo M, Pinzón A, Pritchard L,
				Ramsahoye B, Ren Q, Restrepo S, Roy S, Sadanandom A, Savidor A, Schornack
				S, Schwartz DC, Schumann UD, Schwessinger B, Seyer L, Sharpe T, Silvar C,
				Song J, Studholme DJ, Sykes S, Thines M, van de Vondervoort PJI,
				Phuntumart V, Wawra S, Weide R, Win J, Young C, Zhou S, Fry W, Meyers BC,
				van West P, Ristaino J, Govers F, Birch PRJ, Whisson SC, Judelson HS,
				Nusbaum C. 2009. Genome sequence and analysis of the Irish potato famine
				pathogen <i>Phytophthora infestans</i>. <i>Nature</i> 461:393–398. DOI:
				<a href="https://doi.org/10.1038/nature08358">10.1038/nature08358</a>.</p>
				<p>Fahlgren N, Sullivan CM, Kasschau KD, Chapman EJ, Cumbie JS,
				Montgomery TA, Gilbert SD, Dasenko M, Backman TWH, Givan SA, Carrington
				JC. 2009. Computational and analytical framework for small RNA profiling
				by high-throughput sequencing. <i>RNA</i> 15:992–1002. DOI:
				<a href="https://doi.org/10.1261/rna.1473809">10.1261/rna.1473809</a>.</p>
				<p>Montgomery TA, Yoo SJ, Fahlgren N, Gilbert SD, Howell MD, Sullivan CM,
				Alexander A, Nguyen G, Allen E, Ahn JH, Carrington JC. 2008. AGO1-miR173
				complex initiates phased siRNA formation in plants. <i>Proceedings of the
				National Academy of Sciences of the United States of America</i>
				105:20055–20062. DOI: <a href="https://doi.org/10.1073/pnas.0810241105">
				10.1073/pnas.0810241105</a>.</p>
				<p>Batista PJ, Ruby JG, Claycomb JM, Chiang R, Fahlgren N, Kasschau KD,
				Chaves DA, Gu W, Vasale JJ, Duan S, Conte D Jr, Luo S, Schroth GP,
				Carrington JC, Bartel DP, Mello CC. 2008. PRG-1 and 21U-RNAs interact to
				form the piRNA complex required for fertility in <i>C. elegans</i>.
				<i>Molecular Cell</i> 31:67–78. DOI:
				<a href="https://doi.org/10.1016/j.molcel.2008.06.002">
				10.1016/j.molcel.2008.06.002</a>.</p>
				<p>Montgomery TA, Howell MD, Cuperus JT, Li D, Hansen JE, Alexander AL,
				Chapman EJ, Fahlgren N, Allen E, Carrington JC. 2008. Specificity of
				ARGONAUTE7-miR390 interaction and dual functionality in <i>TAS3</i>
				trans-acting siRNA formation. <i>Cell</i> 133:128–141. DOI:
				<a href="https://doi.org/10.1016/j.cell.2008.02.033">
				10.1016/j.cell.2008.02.033</a>.</p>
				<p>Backman TWH, Sullivan CM, Cumbie JS, Miller ZA, Chapman EJ, Fahlgren
				N, Givan SA, Carrington JC, Kasschau KD. 2008. Update of ASRP: the
				Arabidopsis Small RNA Project database. <i>Nucleic Acids Research</i>
				36:D982–5. DOI: <a href="https://doi.org/10.1093/nar/gkm997">
				10.1093/nar/gkm997</a>.</p>
				<p>Liu P-P, Montgomery TA, Fahlgren N, Kasschau KD, Nonogaki H,
				Carrington JC. 2007. Repression of <i>AUXIN RESPONSE FACTOR10</i> by
				microRNA160 is critical for seed germination and post-germination stages.
				<i>The Plant Journal: for cell and molecular biology</i> 52:133–146. DOI:
				<a href="https://doi.org/10.1111/j.1365-313X.2007.03218.x">
				10.1111/j.1365-313X.2007.03218.x</a>.</p>
				<p>Kasschau KD, Fahlgren N, Chapman EJ, Sullivan CM, Cumbie JS, Givan SA,
				Carrington JC. 2007. Genome-wide profiling and analysis of Arabidopsis
				siRNAs. <i>PLoS Biology</i> 5:e57. DOI:
				<a href="https://doi.org/10.1371/journal.pbio.0050057">
				10.1371/journal.pbio.0050057</a>.</p>
				<p>Howell MD, Fahlgren N, Chapman EJ, Cumbie JS, Sullivan CM, Givan SA,
				Kasschau KD, Carrington JC. 2007. Genome-wide analysis of the
				RNA-DEPENDENT RNA POLYMERASE6/DICER-LIKE4 pathway in Arabidopsis reveals
				dependency on miRNA- and tasiRNA-directed targeting.
				<i>The Plant Cell</i> 19:926–942. DOI:
				<a href="https://doi.org/10.1105/tpc.107.050062">
				10.1105/tpc.107.050062</a>.</p>
				<p>Fahlgren N, Howell MD, Kasschau KD, Chapman EJ, Sullivan CM, Cumbie
				JS, Givan SA, Law TF, Grant SR, Dangl JL, Carrington JC. 2007.
				High-throughput sequencing of Arabidopsis microRNAs: evidence for
				frequent birth and death of <i>MIRNA</i> genes. <i>PloS ONE</i> 2:e219.
				DOI: <a href="https://doi.org/10.1371/journal.pone.0000219">
				10.1371/journal.pone.0000219</a>.</p>
				<p>Fahlgren N, Montgomery TA, Howell MD, Allen E, Dvorak SK, Alexander
				AL, Carrington JC. 2006. Regulation of <i>AUXIN RESPONSE FACTOR3</i> by
				<i>TAS3</i> ta-siRNA affects developmental timing and patterning in
				Arabidopsis. <i>Current Biology</i>: CB 16:939–944. DOI:
				<a href="https://doi.org/10.1016/j.cub.2006.03.065">
				10.1016/j.cub.2006.03.065</a>.</p>
				<p>Xie Z, Allen E, Fahlgren N, Calamar A, Givan SA, Carrington JC. 2005.
				Expression of Arabidopsis <i>MIRNA</i> genes. <i>Plant Physiology</i>
				138:2145–2154. DOI: <a href="https://doi.org/10.1104/pp.105.062943">
				10.1104/pp.105.062943</a>.</p>
			</div>
		</div>
        <div class="row">
        	<div class="col-lg-8 col-lg-offset-2 text-center">
        		<a href="#contact" class="btn btn-circle page-scroll">
            		<i class="fa fa-angle-double-down animated"></i>
        		</a>
        	</div>
        </div>
    </section>
